internal/controllers/topology/cluster/patches/inline/json_patch_generator.go

```diff
@@ -170,9 +171,15 @@ func matchesSelector(req *runtimehooksv1.GeneratePatchesRequestItem, templateVar
 				// If templateMDClass matches one of the configured MachineDeploymentClasses.
 				for _, mdClass := range selector.MatchResources.MachineDeploymentClass.Names {
 					// We have to quote mdClass as templateMDClassJSON is a JSON string (e.g. "default-worker").
-					if string(templateMDClassJSON.Raw) == strconv.Quote(mdClass) {
+					if strconv.Quote(mdClass) == strconv.Quote("*") || string(templateMDClassJSON.Raw) == strconv.Quote(mdClass) {
 						return true
 					}
+					regex, err := regexp.Compile(mdClass)
+					if err == nil {
+						if regex.MatchString(string(templateMDClassJSON.Raw)) {
+							return true
+						}
+					}
```

Comment:
The idea was to use glob matching not regex.

I expected that Go provides a package for that. Apparently there are only filepath.Glob and filepath.Match

Assuming that we don't want to implement glob matching ourselves, did we have any specific library in mind @vincepri @fabriziopandini? (Maybe I'm missing an easy way to implement this).

Btw. I would be absolutely fine with just implementing `*` and not also trying to implement matching for globas in general (like `test-*`)

Reply:
Those are valid concern, I wouldn't expect us to copy the glob package, but we can probably rather start with something simple for the most common use cases. (Solve for the 80% for now)

We should be able to support the following:

- `*` -> everything
- `test-*` -> prefix
- `*-test` -> suffix

A few constraints we should put in place in our own code:

- Every string can at most have a single `*` rune.
- The `*` rune can appear only at the beginning, or ending of the string.
- The rest of the string without the `*` character should comply with Kubernetes naming standards.

Reply:
Sounds good to me

@valaparthvi We discussed what we consider "comply with Kubernetes naming standards" and it comes down to doing this validation:
```go
			// Replace * with the valid character "a" and then validate if name is a qualified name.
			validation.IsQualifiedName(strings.ReplaceAll(name, "*", "a"))
```

Everything below "A few constraints we should put in place in our own code:" should be added to the `validateSelectors` func in our ClusterClass webhook.

Reply:
>The rest of the string without the * character should comply with Kubernetes naming standards.

>We discussed what we consider "comply with Kubernetes naming standards" and it comes down to doing this validation: (+ handle the returned errors)

I am assuming this applies to both strings with/without "*". The first comment mentions without "*", but I think it equally applies to both. Correct me if I am wrong.

Reply:
>The idea was to use glob matching not regex.

There is no built-in glob pkg in Golang, would you like me to use an external pkg or can we make do with regex? TBH, I am not sure I understand glob matching. I understand that it is used for file matching, but why do we need it here to match names?

Reply:
The idea was to have some simple way to match without having to expose full regex support.

The idea now was to do a custom implementation for those rules:
* * -> everything
* test-* -> prefix
* *-test -> suffix

Reply:
>The idea now was to do a custom implementation for those rules: (as there is no standard lib for glob)

I'm going to attempt to write https://go.dev/src/path/filepath/match.go for matching name globs.

Reply:
WDYT about using an existing pkg: https://github.com/gobwas/glob? This pkg is not actively maintained, but it seems [mature](https://github.com/gobwas/glob#performance) and is used by [thousands](https://github.com/gobwas/glob/network/dependents).

Reply:
Personally, I would avoid a new dependency, especially if not maintained.
Also, what we need is just (StartsWith * and HasSuffix ...)  or (EndsWith * and HasPrefix ...), so it seems overkilling

Reply:
> Personally, I would avoid a new dependency, especially if not maintained. Also, what we need is just (StartsWith * and HasSuffix ...) or (EndsWith * and HasPrefix ...), so it seems overkilling

So if you're okay with what we have right now, I can start working on unit tests.